### PR TITLE
Username password auth

### PR DESCRIPTION
**Pull request by beaspider**
I have a use case where I need to authenticate with a username/password from a 3rd party system.
I have added an example username-password-auth

### CODE REVIEW

**Comment by beaspider** [comment]
Have signed the CLA :)

File: username-password-auth/main.css [hunk 1]
```diff
@@ -0,0 +1,62 @@
+/**
+ * Copyright 2015 Google Inc. All Rights Reserved.
```
Comment (nicolasgarnier) [inline username-password-auth/main.css]: I don't think this file is used. remove it? The one that is being served is the main.css in the `public` directory.

File: username-password-auth/public/main.js [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  } else {
+    this.signedOutCard.style.display = 'block';
+    this.signedInCard.style.display = 'none';
+  }
+};
+
+// Initiates the sign-in flow.
+Demo.prototype.signIn = function() {
+    var err = this.signInError;
+    err.innerText = "";
```
Comment (nicolasgarnier) [inline username-password-auth/public/main.js]: Can you use single quotes for Strings in JavaScrip in the whole file?

**Comment by beaspider** [comment]
I have completed all the requested changes + added some extra error handling :)

**Comment by nicolasgarnier** [comment]
Looks good, thanks so much @beaspider !

**Pull request merged**